packages/vm/examples/7702/uniswap-swap-transfer.ts

```diff
@@ -0,0 +1,107 @@
+import { Common, Hardfork, Mainnet } from '@ethereumjs/common'
+import { RPCStateManager } from '@ethereumjs/statemanager'
+import { EOACode7702Tx, type EOACode7702TxData } from '@ethereumjs/tx'
+import type { PrefixedHexString } from '@ethereumjs/util'
+import {
+  createAddressFromPrivateKey,
+  eoaCode7702SignAuthorization,
+  hexToBytes,
+} from '@ethereumjs/util'
+import { createVM, runTx } from '@ethereumjs/vm'
+import { Interface, parseEther, parseUnits } from 'ethers'
+
+async function run() {
+  // ─── your EOA key & address ───────────────────────────────────────────
+  const privateKeyHex = '0x1122334455667788112233445566778811223344556677881122334455667788'
+  const privateKey = hexToBytes(privateKeyHex)
+  const userAddress = createAddressFromPrivateKey(privateKey)
+  console.log('EOA:', userAddress.toString())
+
+  // ─── set up EthereumJS VM with EIP-7702 enabled ───────────────────────
+  const common = new Common({
+    chain: Mainnet,
+    hardfork: Hardfork.Cancun,
+    eips: [7702],
+  })
+  const stateManager = new RPCStateManager({
+    provider: 'YourProviderURLHere',
+    blockTag: 22_000_000n,
+  })
+  const vm = await createVM({ common, stateManager })
+
+  // ─── constants & ABIs ────────────────────────────────────────────────
+  const DAI = '0x6B175474E89094C44Da98b954EedeAC495271d0F'
+  const UNISWAP_V3_ROUTER = '0xE592427A0AEce92De3Edee1F18E0157C05861564'
+  const WETH = '0xC02aaa39b223FE8D0A0e5c4F27EaD9083C756Cc2'
+  const COLD_WALLET = `0x${'42'.repeat(20)}`
+  const BATCH_CONTRACT = '0xYourBatchContractAddressHere'
+
+  const erc20Abi = [
+    'function approve(address _spender, uint256 _amount) external returns (bool success)',
+    'function transfer(address _to, uint256 _value) public returns (bool success)',
+  ]
+  const routerAbi = ['function exactInput(bytes)']
+  const batchAbi = ['function executeBatch(bytes[] calldata) external']
```

Comment:
Oh nvm this interface is here. But it seems that the encoding done in this example is done differently :thinking: